Get_Repo_Details/get_repo_details.py

```diff
@@ -0,0 +1,90 @@
+# Script to GET Repo details from user
+
+import json
+import requests
+import os
+import pprint
+import fire
+
+# constants useful for the script
+BASE_URL = "https://api.github.com/graphql" # Github base url
+GITHUB_TOKEN = os.environ['GITHUB_TOKEN']  # Github token environment variable
+
+def getQuery(repoName, userName):
+  '''
+  Returns the query string that will fetch data from the API.
+  Gets the number of collaborators, number of stars, 
+  number of pull requests, and total commits of a particular repository.
+
+  Args:
+    repoName : Name of the repository from which to fetch data
+    userName : Name of the owner of the repository
+  
+  Return:
+    query : Query string which will be used to query the API endpoint
+  '''
+  query = """query {{
+    repository(name: \"{0}\", owner: \"{1}\") {{
+      collaborators {{
+        totalCount
+      }}
+      pullRequests {{
+        totalCount
+      }}
+      stargazers {{
+        totalCount
+      }}
+      object(expression: "master") {{
+        ... on Commit {{
+          history {{
+            totalCount
+          }}
+        }}
+      }}
+    }}
+  }}""".format(repoName, userName)   # GraphQL Query
+  return query
+
+
+def getRepoDetails(repoName, userName):
+  # Main function
+  # Set names of Repo and gets data from API Query
+
+  # Sanity check
+  assert (repoName is not None and userName is not None), "Invalid Details"
+
+  # Specify Headers
+  # Note - Can only see collaborators if it is own repository
+  headers = {'Authorization': "Token " + GITHUB_TOKEN}
+  # API Request
+  response = requests.post(
+    BASE_URL,
+    json={'query': getQuery(repoName, userName)},
```

Comment:
# GitHub's API endpoint for  GraphQL
url = 'https://api.github.com/graphql'

# Headers for Authentication
headers = {'Authorization': "Token " + access_token}

# Making our API request
r = requests.post(url, json={'query': query}, headers=headers)

# Printing the Status Code to check for any error
print(r.status_code)